pkg/roachprod/vm/gce/gcloud.go

```diff
@@ -1244,8 +1244,9 @@ func (p *Provider) computeInstanceArgs(
 		for i := 0; i < providerOpts.SSDCount; i++ {
 			args = append(args, "--local-ssd", "interface=NVME")
 		}
+		extraMountOpts = "discard"
```

Comment:
Would be nice to include a comment here with a summary of the issue it's solving (and even a reference to that issue). We can salvage that with a git blame, but not a bad idea to have that inline.

Reply:
I think you might be referring to: https://github.com/cockroachdb/cockroach/pull/123050 - which was the disk failure issue?

We didn't really find anything by adding discard, but we did notice the OS was not discarding automatically. A `fstrim` resulted in claiming back huge amounts of free blocks. This has a performance impact "likely" (which we did not measure). But we are adding it for good measure, as we want the disk to collect free blocks in it's free time, rather than having to do it on-demand.

I added the google doc example of mounting NVMe Local SSDs, which implies we should add it.

Reply:
This could _potentially_ cause a perf. regression, so we should probably add a roachperf annotation. Note that while `discard` might be the lesser evil, customers are likely _not_ using it for reasons of performance, e.g., [1].

[1] https://cockroachlabs.slack.com/archives/C06CZB2R7L2/p1707332121425529?thread_ts=1707331431.739319&cid=C06CZB2R7L2

Reply:
Evidently, TRIM isn't needed (i.e., not recommended) on modern SSDs [1].

[1] https://www.spinics.net/lists/raid/msg40916.html